llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -3110,6 +3110,19 @@ KnownBits SelectionDAG::computeKnownBits(SDValue Op, const APInt &DemandedElts,
     }
     break;
   }
+  case ISD::STEP_VECTOR: {
+    const APInt &Step = Op.getConstantOperandAPInt(0);
+
+    if (Step.isPowerOf2())
+      Known.Zero.setLowBits(Step.logBase2());
+
+    const Function &F = getMachineFunction().getFunction();
+    const APInt MaxNumElts = getVScaleRange(&F, BitWidth).getUnsignedMax() *
```

Comment:
Do we need to check for overflow on these multiplies?

Reply:
Yeah, I did a quick check that * does indeed perform wrapping. But forgot to actually bail if it overflows.

Reply:
I think in practice, the overflow will never happen for current targets.  The maximum value of vscale for AArch64 and RISCV is quite small and the maximum size of the vector isn't likely to be much larger than a m8 in practice.  Should definitely guard the case properly for future proofing though.

Reply:
I think it's the multiplicand that's most likely to cause overflow though. On RVV strided ops can have a 2^XLEN - 1 stride max. But as you say, I hope we won't see this in practice.